source/common/event/dispatcher_impl.cc

```diff
@@ -119,9 +119,9 @@ DispatcherImpl::createSslListener(Network::ConnectionHandler& conn_handler,
 }
 
 TimerPtr DispatcherImpl::createTimer(TimerCb cb) {
-  // TODO(htuch): Figure out why we can't set this assertion, see
-  // https://github.com/lyft/envoy/issues/1294.
-  // ASSERT(isThreadSafe());
+  // We don't ASSERT that createTimer() is called on the run thread because timers are used for
```

Comment:
Hah, actually, we don't even need createTimerInternal, we just need to move this init of 0 to before we create the timers: https://github.com/lyft/envoy/blob/master/source/common/event/dispatcher_impl.h#L83 😉